### PR TITLE
Set up GH Actions for CI and doc deployment

### LINKED ISSUE #134: Documentation on website is outdated
https://juliamath.github.io/Primes.jl/stable/ still is for version 0.5.0

This is because there is no working CI workflow set up for updating the website

### PR DESCRIPTION
**Pull request by fingolfin**
Also update TagBot setup

I don't know if the repository settings contain a valid `DOCUMENTER_KEY` in the secrets, as I don't have any permissions on this repository. Someone who has (@oscardssmith ?) needs to [follow these instructions](https://documenter.juliadocs.org/stable/man/hosting/#Authentication:-SSH-Deploy-Keys) if that key is not yet present.

Closes #90
Resolves #134 (hopefully)

### CODE REVIEW

**Comment by oscardssmith** [comment]
will do. thanks!

File: README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Primes.jl
 
-[![Build Status](https://travis-ci.org/JuliaMath/Primes.jl.svg?branch=master)](https://travis-ci.org/JuliaMath/Primes.jl)
-[![Windows Build](https://ci.appveyor.com/api/projects/status/ao64pk44lwo0092r/branch/master?svg=true)](https://ci.appveyor.com/project/ararslan/primes-jl/branch/master)
```
Comment (fingolfin) [inline README.md:4]: This PR removes the Appveyor config and batch. But I just noticed that it still seems to be used. Personally, I find it easier to just have everything in GitHub Actions these days (less things to remember that way), but if you prefer to keep Appveyor, then I can of course adjust this PR accordingly.

Comment (oscardssmith) [reply README.md:4]: I'm fine moving everything to GitHub actions.

Comment (fingolfin) [reply README.md:4]: Then AppVeyor needs to be removed from branch protection rules.

I think we need to merge this PR to find out if it works, as GH will not run the actions in here if they aren't also on the master branch. Subsequent fixes can then be tested in PRs. I.e. if this fails after merging I can submit a follow-up PR to get it fixed.

**Comment by oscardssmith** [comment]
lets see if this works

**Comment by fingolfin** [comment]
Unfortunately something is wrong. I'll look into it (but later, busy with some other things right now)

**Pull request merged**